packages/renderer/src/lib/table/Table.svelte

```diff
@@ -0,0 +1,148 @@
+<script lang="ts">
+import Checkbox from '../ui/Checkbox.svelte';
+import type { Column, Row } from './table';
+import { flip } from 'svelte/animate';
+
+export let kind: string;
+export let data: any[];
+export let columns: Column<any>[];
+export let row: Row<any>;
+
+let gridCol: string;
+$: gridCol = gridCols();
+
+// number of selected items in the list
+export let selectedItemsNumber: number = 0;
+$: selectedItemsNumber = row.selectable
+  ? data.filter(object => row.selectable(object)).filter(object => object.selected).length
+  : 0;
+
+// do we need to unselect all checkboxes if we don't have all items being selected ?
+$: selectedAllCheckboxes = row.selectable
+  ? data.filter(object => row.selectable(object)).every(object => object.selected)
+  : false;
+
+function toggleAll(checked: boolean) {
+  if (!row.selectable) {
+    return;
+  }
+  const toggleData = data;
+  toggleData.filter(object => row.selectable(object)).forEach(object => (object.selected = checked));
+  data = toggleData;
+}
+
+let sortCol: Column<any>;
+let sortAscending: boolean;
+
+function sort(column: Column<any>) {
+  if (!column) {
+    return;
+  }
+
+  let comparator = column.comparator;
+  if (!comparator) {
+    // column is not sortable
+    return;
+  }
+
+  if (sortCol === column) {
+    sortAscending = !sortAscending;
+  } else {
+    sortCol = column;
+    sortAscending = true;
+  }
+  if (!sortAscending) {
+    // we're already sorted, switch to reverse order
+    let comparatorTemp = comparator;
+    comparator = (a, b) => -comparatorTemp(a, b);
+  }
+  const sortedData = data;
+  sortedData.sort(comparator);
+  data = sortedData;
+}
+
+function gridCols(): string {
+  // section and checkbox columns
+  let columnWidths: string[] = ['20px', '40px'];
+
+  // custom columns
+  for (const column of columns) {
+    if (column.info.width) {
+      columnWidths.push(column.info.width);
+    } else {
+      columnWidths.push('1fr');
+    }
+  }
+
+  return columnWidths.join(',');
+}
+</script>
+
+<div class="w-full" class:hidden="{data.length === 0}" role="table">
+  <!-- Table header -->
+  <div
+    class="grid grid-flow-row grid-cols-[{gridCol}] gap-x-0.5 mx-5 h-7 sticky top-0 bg-charcoal-700 text-xs text-gray-600 font-bold uppercase z-[2]"
```

Comment:
You cannot do `grid-cols-[{gridCol}]` or the style will not be applied. You always have to use static class https://tailwindcss.com/docs/content-configuration#dynamic-class-names
You have to pass the class already defined or think some other way to do it

before
![image](https://github.com/containers/podman-desktop/assets/49404737/bf95c63d-8b10-44bd-9a12-d6def96c892d)
after
![image](https://github.com/containers/podman-desktop/assets/49404737/310a0fbe-0636-4654-9024-9b846170e585)

Reply:
Thank you! I was getting really confused because the behaviour was inconsistent, and had forgotten this should not work. Latest commit has a fix setting the style directly on the element.